lessons/intro/click/index.md

```diff
@@ -188,8 +196,42 @@ def cat(files):
         print(file.read(), end='')
 ```
 
-Existuje i varianta [`click.Path()`](http://click.pocoo.org/6/api/#click.Path),
-která soubor neotvírá. Pomocí ní jde např. zadat jméno adresáře.
+Existuje i varianta [`click.Path()`](https://click.palletsprojects.com/en/7.x/api/#click.Path),
+která soubor neotvírá. Pomocí ní jde např. zadat jméno adresáře. Click takto 
+poskytuje i jiné [další typy](https://click.palletsprojects.com/en/7.x/api/#types).
+
+
+## Validace vstupů
+
+Vstupy získané z přepínačů i argumentů lze ověřit pomocí 
+vlastních podmínek a podle toho naprogramovat chování včetně
+chybových hlášek. Click však opět nabízí pohodlnější způsob, 
+a to pomocí [`callback`](https://click.palletsprojects.com/en/7.x/options/#callbacks-for-validation).
+V rámci callback funkce můžete ověřit libovolně hodnotu a/nebo
+ji vhodně transformovat. Pokud hodnota neodpovídá požadavkům, 
+můžete použít vyjímku [`click.UsageError`](https://click.palletsprojects.com/en/7.x/api/#click.UsageError)
+nebo [`click.BadParameter`](https://click.palletsprojects.com/en/7.x/api/#click.BadParameter)
+(vztahuje-li se přímo ke konkrétnímu parametru). Click se pak 
+sám postará o případné ukončení programu s odpovídající chybovou
+hláškou a kódem.
+
+```python
+def validate_username(ctx, param, value):
+    try:
+        assert len(value) == 8
```

Comment:
Not important, but some usernames can be shorter.

```suggestion
        assert 2 <= len(value) <= 8
```

Reply:
Fixed in https://github.com/cvut/naucse.python.cz/pull/12/commits/5f3035c992fc3bf58ed47b3489f4ebef0e45505f